lib/blockscore/collection.rb

```diff
@@ -133,8 +133,7 @@ def create(params = {})
     # @api public
     def retrieve(id)
       each do |item|
-        next unless item.id == id
-        return item
+        return item if item.id.eql?(id)
```

Comment:
If `id` is an integer, then you could use: `item.id.equal?(id)` since it's a stronger match.